common/components/common/selection/SelectorCollapsible.jsx

```diff
@@ -142,14 +145,44 @@ class SelectorCollapsible<T> extends React.PureComponent<Props<T>, State> {
       <span className="CollapsibleCategorySubheader" onClick={this.expandCategory.bind(this, category)}>
         {category} { } {isExpanded ? this.constants.chevronDown : this.constants.chevronRight}</span>
         { isExpanded
-          ? <ContextualCollapsible>
+          ? <span>
+            {this._renderSelectAll(this.state.optionCategoryTree[category])}
+            <ContextualCollapsible>
               { this._renderOptions(this.state.optionCategoryTree[category]) }
             </ContextualCollapsible>
+            </span>
           : null
         }
       </div>
     });
   }
+  _renderSelectAll(category): React$Node {
+    return (
+      <span>
+        <label className="CollapsibleMenuItem">
+          <
+            input type="checkbox"
+            className="ContextualCollapsible-selectAll"
+            checked={this.props.isAllChecked}
+            onChange={this._handleSelectAll.bind(this,category)}
+          >
+          </input>Select All
+        </label>
+      </span>
+    )
+  }
+
+  _handleSelectAll(category): React$Node {
+    //this feels like a mess, but find what tags are selected or not and then either select or unselect the remainder
+    if(this.props.isAllChecked) {
+      let toUpdate = category.filter(tag => this.props.optionEnabled(tag) === true)
+      this.selectOption(toUpdate, {multiple: true, type: "REMOVE_MANY_TAGS"} );
+    } else {
+      let toUpdate = category.filter(tag => this.props.optionEnabled(tag) === false)
+      this.selectOption(toUpdate, {multiple: true, type: "ADD_TAG"} );
+    }
+    this.props.AllChecked();
```

Comment:
instead of calling this manually I think I need it to be part of an onchange event, right now it's always one interaction behind. if you manually start checking items one at a time and then look at the console you'll see for example it will say items checked 1 when you've just checked the 2nd item in a category